src/Value.ml

```diff
@@ -30,6 +32,7 @@ let rec to_string : t -> string = function
   | Bool b   -> Bool.to_string b
   | Char c   -> "\'" ^ (Char.to_string c) ^ "\'"
   | String s -> "\"" ^ s ^ "\""
+  | Real r   -> Float.(if r < 0. then "(" ^ (to_string r) ^ ")" else to_string r)
   | List _   -> raise (Internal_Exn "List type (de/)serialization not implemented!")
```

Comment:
I don't understand this.  Why do you put parens around it if it's negative?

Reply:
I saw that Saswat used this in the to_string for his num type and thought that it may make things more clear/easy to read when negative reals are printed. Do you think I should always print reals without parentheses?

Reply:
My bad -- I wasn't super careful here. In that project we don't deserialize serialized values, only the input spreadsheet. Here we would need to deserialize serialized states and counterexamples etc.

The `of_string` for `Num` seems incompatible with `to_string`. I'd say remove the parens here.

Reply:
I see, thanks for the explanation. Looks like I was following your code a bit too closely.